### PR TITLE
Add storage load/store functions to Yul dialect.

### PR DESCRIPTION
**Pull request by ekpyron**
@chriseth is this about the amount you want to have extracted here: https://github.com/ethereum/solidity/pull/10467#discussion_r534035494 ?
Or are the casted-enum-index-lookups into the member arrays already a bit over the top? If so I can dial it down a bit:-)...

### CODE REVIEW

File: libyul/optimiser/DataFlowAnalyzer.cpp [hunk 1]
```diff
@@ -391,53 +401,27 @@ bool DataFlowAnalyzer::inScope(YulString _variableName) const
 }
 
 std::optional<pair<YulString, YulString>> DataFlowAnalyzer::isSimpleStore(
-	evmasm::Instruction _store,
```
Comment (axic) [inline libyul/optimiser/DataFlowAnalyzer.cpp:394]: Can you remove the includes for `libevmasm` then? Or is it still needed?

Comment (ekpyron) [reply libyul/optimiser/DataFlowAnalyzer.cpp:394]: There was even a fat ``TODO avoid`` comment there and I still missed it :-). Removed it now (and the EVMDialect include in the implementation as well).

File: libyul/optimiser/DataFlowAnalyzer.h [hunk 2]
```diff
@@ -29,9 +29,6 @@
 #include <libyul/AST.h> // Needed for m_zero below.
 #include <libyul/SideEffects.h>
 
-// TODO avoid
-#include <libevmasm/Instruction.h>
```
Comment (axic) [inline libyul/optimiser/DataFlowAnalyzer.h:33]: Also LoadResolver.h has it.

**Comment by ekpyron** [comment]
Actually I embarrassingly also forgot to actually fill the members in the ``DataFlowAnalyzer`` constructor... upcoming in a second...

**Comment by ekpyron** [comment]
I just force-pushed them in now, I hope that's fine.

**Review by chriseth — APPROVED** [review]

**Pull request merged**